framework/auth/campaigns.py

```diff
@@ -1,68 +1,98 @@
-import httplib as http
-
 import furl
+
 from werkzeug.datastructures import ImmutableDict
-from framework.exceptions import HTTPError
 
 from website import mails
 from website.settings import DOMAIN
 
-CAMPAIGNS = ImmutableDict({
+CAMPAIGNS = {
     'prereg': {
         'system_tag': 'prereg_challenge_campaign',
-        'redirect_url': lambda: furl.furl(DOMAIN).add(path='prereg/').url,
+        'redirect_url': furl.furl(DOMAIN).add(path='prereg/').url,
         'confirmation_email_template': mails.CONFIRM_EMAIL_PREREG,
     },
     'institution': {
         'system_tag': 'institution_campaign',
-        'redirect_url': lambda: ''
+        'redirect_url': ''
     },
     'erpc': {
         'system_tag': 'erp_challenge_campaign',
-        'redirect_url': lambda: furl.furl(DOMAIN).add(path='erpc/').url,
+        'redirect_url': furl.furl(DOMAIN).add(path='erpc/').url,
         'confirmation_email_template': mails.CONFIRM_EMAIL_ERPC,
     },
     # Various preprint services
     # Each preprint service will offer their own campaign with appropriate distinct branding
     'osf-preprints': {
         'system_tag': 'osf_preprints',
-        'redirect_url': lambda: furl.furl(DOMAIN).add(path='preprints/').url,
-        'confirmation_email_template': mails.CONFIRM_EMAIL_PREPRINTS_OSF,
+        'redirect_url': furl.furl(DOMAIN).add(path='preprints/').url,
+        'confirmation_email_template': mails.CONFIRM_EMAIL_PREPRINTS('osf', 'OSF'),
+        'proxy_login': True,
+        'branded': False,
+        'provider': 'OSF'
+    },
+}
+
+providers = [
+    'SocArXiv',
+    'engrXiv',
+    'PsyArXiv'
+]
+
+for provider in providers:
+    provider_id = provider.lower()
+    key = '{}-preprints'.format(provider_id)
+    tag = '{}_preprints'.format(provider_id)
+    path = 'preprints/{}'.format(provider_id)
+    CAMPAIGNS[key] = {
+        'system_tag': tag,
+        'redirect_url': furl.furl(DOMAIN).add(path=path).url,
+        'confirmation_email_template': mails.CONFIRM_EMAIL_PREPRINTS('branded', provider),
         'proxy_login': True,
+        'branded': True,
+        'provider': provider,
     }
-})
+
+CAMPAIGNS = ImmutableDict(CAMPAIGNS)
 
 
 def system_tag_for_campaign(campaign):
     if campaign in CAMPAIGNS:
-        return CAMPAIGNS[campaign]['system_tag']
+        return CAMPAIGNS[campaign].get('system_tag')
```

Comment:
Update all campaign functions in the same format and use `.get()` instead for `[]`.
```python
if campaign in CAMPAIGNS:
    # do the job
return None
```